src/assets/TokenRatesController.ts

```diff
@@ -54,9 +73,34 @@ export interface TokenRatesConfig extends BaseConfig {
  * Token rates controller state
  *
  * @property contractExchangeRates - Hash of token contract addresses to exchange rates
+ * @property supportedChains - Cached chain data
  */
 export interface TokenRatesState extends BaseState {
-  contractExchangeRates: { [address: string]: number };
+  contractExchangeRates: ContractExchangeRates;
+  supportedChains: ChainCaches;
+}
+
+const COINGECKO_API = {
+  BASE_URL: 'https://api.coingecko.com/api/v3',
+  getTokenPriceURL(chainSlug: string, query: string) {
+    return `${this.BASE_URL}/simple/token_price/${chainSlug}?${query}`;
+  },
+  getPlatformsURL() {
+    return `${this.BASE_URL}/asset_platforms`;
+  },
+};
+
+function getUpdatedSupportedChains(
```

Comment:
This function seems.... odd. It just combines the two parameters? Seems to have limited utility at first glance.

Reply:
Yeah, it actually merges a nested property of state because `BaseController.update` is shallow merge, would override other `chainIds` values.